apps/rush-lib/src/cli/actions/GenerateAction.ts

```diff
@@ -105,20 +108,27 @@ export default class GenerateAction extends BaseRushAction {
       // Do an incremental install
       installManager.installCommonModules(InstallType.Normal);
 
-      console.log(os.EOL + colors.bold('(Skipping "npm shrinkwrap")') + os.EOL);
+      if (this.rushConfiguration.packageManager === 'npm') {
+        console.log(os.EOL + colors.bold('(Skipping "npm shrinkwrap")') + os.EOL);
```

Comment:
>console.log(os.EOL + colors.bold('(Skipping "npm shrinkwrap")') + os.EOL); [](start = 8, length = 74)

There's an NPM 5 bug here.

NPM 4 by default doesn't create a shrinkwrap file, so we had to run "npm shrinkwrap".  This code is skipping that for a lazy install.

NPM 5 and PNPM both DO create a shrinkwrap file.  So:

1. We should really NOT be running "npm shrinkwrap" for a normal generate
2. For a lazy generate, we should be deleting the autogenerated file for both NPM 5 and PNPM.

Also, for PNPM the "--lazy" behavior could be the default.

Reply:
Maybe this logic needs to be pushed down into the subclass?

---
In reply to: [151782133](https://github.com/Microsoft/web-build-tools/pull/427#discussion_r151782133) [](ancestors = 151782133)